src/main/java/cz/fi/muni/pa165/Model/Entity/ApplicationApprovedRecord.java

```diff
@@ -59,4 +60,30 @@ public Date getTo() {
     public RentApplication getApplication() {
         return application;
     } 
+    
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if ((o == null) || !(o instanceof ApplicationApprovedRecord)) {
+            return false;
+        }
+        ApplicationApprovedRecord other = (ApplicationApprovedRecord) o;
+        if (getFrom() != null ? !getFrom().equals(other.getFrom()) : other.getFrom() != null) {
+            return false;
+        }
+        if (getTo() != null ? !getTo().equals(other.getTo()) : other.getTo() != null) {
+            return false;
+        }
+        if (application != null ? !application.equals(other.getApplication()) : other.getApplication() != null) {
+            return false;
+        }
+	return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder().append(getFrom()).append(getTo()).append(getApplication().getId()).toHashCode();
```

Comment:
@charlliz read my comment again and then read definiton of entity.

Two entities in two different states with the same ID are referring to the same real-world object. Thus are the same. **Please fix this by sending another PR.**

@rtrembecky Please, do NOT merge pull's until discussion is over!